src/main/java/Duke.java

```diff
@@ -1,10 +1,423 @@
+
+    /** User commands available to the user */
+    private static final String CMD_ADD_TODO = "todo";
+    private static final String CMD_ADD_DEADLINE = "deadline";
+    private static final String CMD_ADD_EVENT = "event";
+    private static final String CMD_DONE = "done";
+    private static final String CMD_LIST = "list";
+    private static final String CMD_EXIT = "bye";
+    private static final String CMD_HELP = "help";
+    private static final String CMD_CLEAR_WINDOW = "clear";
+
+    /** Easter egg */
+    private static final String CMD_DEBUG = "debug";
+
+    /** Flag to check whether the main program should be running or not */
+    private static boolean isProgramRunning = true;
+
+    /** Flag to avoid printing unnecessary output when debugging */
+    private static boolean inDebugMode = false;
+
+
+    /**
+     * Prints horizontal line for chat bot.
+     */
+    public static void printLine (boolean hasNewline) {
+        System.out.println("  _____________________________________________________________________");
+
+        if (hasNewline) {
+            System.out.println();
+        }
     }
+
+    /**
+     * Prints logo for the bot.
+     */
+    public static void printLogo () {
+
+        String logo = "\t ________  ________  ________  ________  _________  ________     \n" +
+                "\t|\\_____  \\|\\   __  \\|\\   __  \\|\\   __  \\|\\___   ___\\\\   __  \\    \n" +
+                "\t \\|___/  /\\ \\  \\|\\  \\ \\  \\|\\  \\ \\  \\|\\  \\|___ \\  \\_\\ \\  \\|\\  \\   \n" +
+                "\t     /  / /\\ \\   __  \\ \\   ____\\ \\   __  \\   \\ \\  \\ \\ \\  \\\\\\  \\  \n" +
+                "\t    /  /_/__\\ \\  \\ \\  \\ \\  \\___|\\ \\  \\ \\  \\   \\ \\  \\ \\ \\  \\\\\\  \\ \n" +
+                "\t   |\\________\\ \\__\\ \\__\\ \\__\\    \\ \\__\\ \\__\\   \\ \\__\\ \\ \\_______\\\n" +
+                "\t    \\|_______|\\|__|\\|__|\\|__|     \\|__|\\|__|    \\|__|  \\|_______|\n" +
+                "\t                                                                 ";
+
+        System.out.println(logo + "\n");
+    }
+
+    /**
+     * Greets user.
+     */
+    public static void greetUser () {
+
+        if (!inDebugMode) {
+            String name = "Zapato";
+
+            printLine(false);
+            printLogo();
+
+            System.out.println("\tHello from " + name + ":)");
+            System.out.println("\tWhat can I do for you? (type \"help\" for more info)");
+            printLine(true);
+        }
+
+    }
+
+    /**
+     * Displays farewell message.
+     */
+    public static void displayFarewell () {
+        replayBack("Bye! Hope to see you soon :)");
+    }
+
+    /**
+     * Repeats whatever message it receives.
+     *
+     * @param msg Message to print.
+     */
+    public static void replayBack (String msg) {
+
+        printLine(false);
+        System.out.println("\t" + msg);
+        printLine(true);
+    }
```

Comment:
Maybe consider changing the name to replyToUser()? I was confused at first at the purpose of the function based on its name

Reply:
Thank you :) I will make the change before I make my next push. The replyToUser() name is more clear